smartcontractkit/external-adapters-js
# Augment MLB endpoint for SportsDataIO

## Closes20264

## Description

Add additional capabilities to the MLB SportsDataIO endpoint.

## Changes

Augments the MLB endpoint for SportsDataIO with the following

 - Allow filtering game schedule by `teamID`
 - Add `onlyShowGameIDs` flag to `schedule` endpoint to only have `gameIDs` in the response.
 - Set `AwayTeamMoneyLine`, `HomeTeamMoneyLine`, `AwayTeamRuns` and `HomeTeamRuns` to `-1` if a the game's status is not `Final`.

## Steps to Test

1. Run adapter locally and test using the below request payloads and verify responses.

Filter by team
```
{
  "id": 1,
  "data": {
    "date": "2017-JUL-31",
    "sport": "mlb",
    "endpoint": "schedule",
    "teamID": "BAL"
    }
}
```

Only show game IDs

```
{
  "id": 1,
  "data": {
    "date": "2017-JUL-31",
    "sport": "mlb",
    "endpoint": "schedule",
    "onlyShowGameIDs": true
  }
}
```

## Quality Assurance
